flink-filesystems/flink-azure-fs-hadoop/src/main/java/org/apache/flink/fs/azurefs/AbstractAzureFSFactory.java

```diff
@@ -48,6 +55,33 @@ public void configure(Configuration config) {
 	public FileSystem create(URI fsUri) throws IOException {
 		checkNotNull(fsUri, "passed file system URI object should not be null");
 		LOG.info("Trying to load and instantiate Azure File System");
-		return new AzureFileSystem(fsUri, flinkConfig);
+		return new HadoopFileSystem(createInitializedAzureFS(fsUri, flinkConfig));
+	}
+
+	// uri is of the form: wasb(s)://yourcontainer@youraccount.blob.core.windows.net/testDir
+	private org.apache.hadoop.fs.FileSystem createInitializedAzureFS(URI fsUri, Configuration flinkConfig) throws IOException {
+		org.apache.hadoop.conf.Configuration hadoopConfig = HadoopUtils.getHadoopConfiguration(flinkConfig);
```

Comment:
Can't we use the `HadoopConfigLoader` here? Then we would not have to do the manual copying step.

Reply:
Yeah I can use the HadoopConfigLoader. It is a slightly more involved change though as the config loader needs to be moved out of the s3-base module to a new one and we also have other file system factories doing similar copying which also need to be updated if we want to do things thoroughly. 
If you're keen on using it here as part of this PR, I could kick start it by creating a filesystem commons module, move the config loader there and use it in this factory. wdyt?

Reply:
Looking at the details of the `HadoopConfigLoader` and the `HadoopUtils`, it actually makes a bigger difference than thought. `HadoopUtils` loads the Hadoop configuration from a set of specified locations (assuming that Hadoop is available). In contrast to that `HadoopConfigLoader` creates a vanilla `HadoopConfiguration` and populates it with the contents of the Flink configuration. So the `HadoopConfigLoader` is actually used in the S3 FileSystem implementations to completely encapsulate the Hadoop dependencies so that the S3 filesystem implementation is completely self-contained (see FLINK-10362). I think we should do the same for the Azure FS since it is not relevant for the user which implementation we use here (Hadoop) and one should also be able to use it without having Hadoop installed.

I would propose to move the `HadoopConfigLoader` to the `flink-hadoop-fs` module since all implementations which use a Hadoop based implementation will depend on this one. I think the only filesystem not doing it correctly at the moment is the `SwiftFileSystem`. Once we have moved the utility, we should create a JIRA issue to update this file system as a follow up.

Reply:
Ok, sounds good. I'll take a stab at this and can create a follow up jira for the swiftfs.

Reply:
Updated the review to incorporate this. I've filed: https://jira.apache.org/jira/browse/FLINK-12394